modules/reindex/src/test/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSourceTests.java

```diff
@@ -429,6 +431,44 @@ public void testWrapExceptionToPreserveStatus() throws IOException {
         assertEquals(badEntityException, wrapped.getSuppressed()[0]);
     }
 
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public void testTooLargeResponse() throws Exception {
+        ContentTooLongException tooLong = new ContentTooLongException("too long!");
+        CloseableHttpAsyncClient httpClient = mock(CloseableHttpAsyncClient.class);
+        when(httpClient.<HttpResponse>execute(any(HttpAsyncRequestProducer.class), any(HttpAsyncResponseConsumer.class),
+                any(FutureCallback.class))).then(new Answer<Future<HttpResponse>>() {
+                    @Override
+                    public Future<HttpResponse> answer(InvocationOnMock invocationOnMock) throws Throwable {
+                        HeapBufferedAsyncResponseConsumer consumer = (HeapBufferedAsyncResponseConsumer) invocationOnMock.getArguments()[1];
+                        FutureCallback callback = (FutureCallback) invocationOnMock.getArguments()[2];
+
+                        assertEquals(200 * 1024 * 1024, consumer.getBufferLimit());
+                        callback.failed(tooLong);
+                        return null;
+                    }
+                });
+        RemoteScrollableHitSource source = sourceWithMockedClient(true, httpClient);
+
+        AtomicBoolean called = new AtomicBoolean();
+        Consumer<Response> checkResponse = r -> called.set(true);
+        Throwable e = expectThrows(RuntimeException.class,
+                () -> source.doStartNextScroll(FAKE_SCROLL_ID, timeValueMillis(0), checkResponse));
+        // Unwrap the some artifacts from the test
+        while (e.getMessage().equals("failed")) {
+            e = e.getCause();
+        }
+        try {
+            assertFalse(called.get());
+            // This next exception is what the user sees
+            assertEquals("Remote responded with a chunk that was too large. Use a smaller batch size.", e.getMessage());
+            // And that exception is reported as being caused by the underlying exception returned by the client
+            assertSame(tooLong, e.getCause());
+        } catch (AssertionError ae) {
```

Comment:
I really don't think we should capture assertions and rethrow, it leaves us really open to typoing and swallowing test failures because we accidentally forgot to rethrow. I assume your intent here was to give more info if one of them failed? In that case, I think ordering them by most to least information would be better (though optional, I mostly care about removing anything catching asserts):

```java
assertThat(e.getMessage(), containsString("Remote responded with a chunk that was too large. Use a smaller batch size."));
assertSame(tooLong, e.getCause());
assertFalse(called.get());
```

And then removing the try/catch altogether.